src/DurableTask.Netherite/Scaling/AzureTableLoadPublisher.cs

```diff
@@ -13,104 +16,84 @@ namespace DurableTask.Netherite.Scaling
 
     class AzureTableLoadPublisher : ILoadPublisherService
     {
-        readonly CloudTable table;
+        readonly TableClient table;
         readonly string taskHubName;
 
         public AzureTableLoadPublisher(string connectionString, string tableName, string taskHubName)
         {
-            var account = CloudStorageAccount.Parse(connectionString);
-            this.table = account.CreateCloudTableClient().GetTableReference(tableName);
+            this.table = new TableClient(connectionString, tableName); 
             this.taskHubName = taskHubName;
         }
 
         public TimeSpan PublishInterval => TimeSpan.FromSeconds(10);
 
         public async Task DeleteIfExistsAsync(CancellationToken cancellationToken)
         {
-            if (! await this.table.ExistsAsync().ConfigureAwait(false))
+            try
             {
-                return;
+                var tableBatch = new List<TableTransactionAction>();
+                await foreach (var e in this.table.QueryAsync<PartitionInfoEntity>($"PartitionKey eq '{this.taskHubName}'"))
```

Comment:
Consider expression, also pass in cancellation:

```suggestion
                await foreach (var e in this.table.QueryAsync<PartitionInfoEntity>(x => x.PartitionKey == this.taskhubName, cancellationToken: cancellationToken)
```

Reply:
I did not know this was supported, will definitely use that (as long as we can be sure the query is compiled correctly).